docs/nia cts multiple source input configuration updates (#12158)
hashicorp/consul: website/content/docs/nia/terraform-modules.mdx

### Source Input

> CTS monitors defined objects to provide source input to a task. This source input satisfies the requirements defined by the task's Terraform Module's [input variables](https://www.terraform.io/docs/language/values/variables.html).

Reviewer: I'm not sure I understand what is being stated here.  What are object in reference to?

Author: Thanks for asking! The objects refer to the Consul objects that CTS monitors for change. In the case of source input, these Consul objects are converted into variables that are needed by the task's Terraform Module. Maybe I could change it to:

> A task monitors Consul objects that are specified by the task's configuration. The Consul objects can be used for the source input that satisfies the requirements defined by the task's Terraform Module's input variables.

Welcome your input!

Author: Yea - as of now Consul object can be services instances, service registration, or consul-kv pair. We're planning to add additional "objects". Open to new terminology!

Reviewer: Gotcha, that helps a lot. What we can do is add an entry in the [Consul glossary page](https://www.consul.io/docs/install/glossary). Then when we use the term Consul objects we can link to that spot in the glossary. At least the first time we use the term.

Author: Interesting! What do you think about adding the entry in the [CTS glossary section](https://www.consul.io/docs/nia#glossary) instead? I think this terminology will only be used by CTS so was thinking about avoiding impacting Consul docs. I'll defer to your thoughts!

Author: Oh, I see. That's good to know. What are your thoughts on:
- I remove "Consul" from the sentence. Thinking about this some more, we may expand beyond monitoring Consul. It might be better to be better to leave object more generic. (Not sure if this reduces reader confusion)
- I create a task to combine the CTS glossary with Consul's glossary. I can raise it with my team in our next week's meeting.

Reviewer: At the very least, these glossary items should be in the main glossary page. As for leaving it generic, that's what is confusing. As a practitioner, I should at least have an idea of what these objects are. Otherwise, it hard for me to understand how to leverage the configuration options if I don't know what is in scope.

Author: Thanks for sharing the perspective on leaving it generic. I'll keep "Consul". I'd like to initially add the "Consul object" terminology in the CTS glossary section for this PR. Then have a separate PR to combine the CTS glossary over to the Consul glossary.

I'd like to do a separate PR since I don't think the glossary will be a straight copy-over. Many of the entries are written within the context of CTS and will need some rework. For example, the entry for "Terraform Module" was intended to give CTS readers a quick definition, but it could look awkward as is in the Consul glossary.

> A task's source input slightly defers the task's condition even though both monitor defined objects. The task's condition monitors defined objects with a configured criteria. When this criteria is satisfied, the task will trigger.

Reviewer: ```suggestion
A task's source input is slightly different from the task's condition, even though both monitor defined objects. The task's condition monitors defined objects with a configured criteria. When this criteria is satisfied, the task will trigger.
```

> The source input however, monitors defined objects with the intent of providing values or metadata about these objects to the Terraform Module. The monitored source input and condition objects can be the same object, such as a task is configured with a `condition "services"` block and `source_includes_var` is set to `true`. The source input and condition can be different objects and configured separately, such as a task configured with a `condition "catalog-services` and `source_input "consul-kv"` block. In this way the source input does not need to be tied to the provided condition in order to satisfy the Terraform Module.

Reviewer: ```suggestion
The source input however, monitors defined objects with the intent of providing values or metadata about these objects to the Terraform Module. The monitored source input and condition objects can be the same object, such as a task is configured with a `condition "services"` block and `source_includes_var` is set to `true`. The source input and condition can be different objects and configured separately, such as a task configured with a `condition "catalog-services` and `source_input "consul-kv"` block. This results in the monitored source input not needing to be tied to the provided condition in order to satisfy the Terraform Module.
```
Does the change still make sense? I tried to clean up the language.

Author: Yes! Makes sense, thank you

> Each type of monitored object can only be defined through one configuration means for a task. For example, if a task monitors services, the task cannot have both `condition "services"` and `source_input "services"` configured. See [Task Source Input configuration](/docs/nia/configuration#task-source-input) for more details.

Reviewer: ```suggestion
Each type of monitored object can only be defined through one configuration definiton of a task. For example, if a task monitors services, the task cannot have both `condition "services"` and `source_input "services"` configured. See [Task Source Input configuration](/docs/nia/configuration#task-source-input) for more details.
```

CTS monitors defined objects to provide source input to a task. This source input satisfies the requirements defined by the task's Terraform Module's [input variables](https://www.terraform.io/docs/language/values/variables.html).

> A task's source input is slightly different from the task's condition, even though both monitor defined objects. The task's condition monitors defined objects with a configured criteria. When this criteria is satisfied, the task will trigger.

Reviewer 2: It's confusing to me that the source input is being compared to the task condition because we haven't defined "task condition" in this topic. Can we assume that readers have read through the other topics in this section and understand the condition concept?

Author: At this point, I would guess it's likely that a user has heard of a task condition and should have some understanding of it. If I were to rate CTS topics, the details of source input are a little deeper than condition. Let me know your perspective. If it's confusing, happy to remove this paragraph if it's detracting readers and not adding value.

> The source input however, monitors defined objects with the intent of providing values or metadata about these objects to the Terraform Module. The monitored source input and condition objects can be the same object, such as a task is configured with a `condition "services"` block and `source_includes_var` is set to `true`. The source input and condition can be different objects and configured separately, such as a task configured with a `condition "catalog-services` and `source_input "consul-kv"` block. This results in the monitored source input not needing to be tied to the provided condition in order to satisfy the Terraform Module.

Reviewer 2: ```suggestion
The source input, however, monitors defined objects with the intent of providing values or metadata about these objects to the Terraform module. The monitored source input and condition objects can be the same object, such as a task configured with a `condition "services"` block and `source_includes_var` set to `true`. The source input and condition can also be different objects and configured separately, such as a task configured with a `condition "catalog-services` and `source_input "consul-kv"` block. As a result, the monitored source input is decoupled from the provided condition in order to satisfy the Terraform module.
```
I don't believe that we capitalize "module" as a proper noun.

> Each type of monitored object can only be defined through one configuration definiton of a task. For example, if a task monitors services, the task cannot have both `condition "services"` and `source_input "services"` configured. See [Task Source Input configuration](/docs/nia/configuration#task-source-input) for more details.

Reviewer 2: ```suggestion
Each type object that CTS monitors can only be defined through one task definition configuration. For example, if a task monitors services, the task cannot have both `condition "services"` and `source_input "services"` configured. See [Task Source Input configuration](/docs/nia/configuration#task-source-input) for more details.
```
Is that clearer/correct?

Author: Ah, it's made me realize how unclear I was. I was Maybe:
```
Each type object that CTS monitors can only be defined through one configuration within a task definition.
```
